x-pack/plugins/synthetics/public/apps/synthetics/components/monitor_add_edit/form/field_config.tsx

```diff
@@ -0,0 +1,979 @@
+    }),
+    showWhen: ['isTLSEnabled', true],
+  },
+  [ConfigKey.TLS_KEY]: {
+    fieldKey: ConfigKey.TLS_KEY,
+    component: EuiTextArea,
+    label: i18n.translate('xpack.synthetics.monitorConfig.clientKey.label', {
+      defaultMessage: 'Client key',
+    }),
+    helpText: i18n.translate('xpack.synthetics.monitorConfig.clientKey.helpText', {
+      defaultMessage: 'PEM formatted certificate key for TLS client authentication.',
+    }),
+    showWhen: ['isTLSEnabled', true],
+  },
+  [ConfigKey.TLS_KEY_PASSPHRASE]: {
+    fieldKey: ConfigKey.TLS_KEY_PASSPHRASE,
+    component: EuiFieldPassword,
+    label: i18n.translate('xpack.synthetics.monitorConfig.clientKeyPassphrase.label', {
+      defaultMessage: 'Client key passphrase',
+    }),
+    helpText: i18n.translate('xpack.synthetics.monitorConfig.clientKeyPassphrase.helpText', {
+      defaultMessage: 'Certificate key passphrase for TLS client authentication.',
+    }),
+    showWhen: ['isTLSEnabled', true],
+  },
+  [ConfigKey.SCREENSHOTS]: {
+    fieldKey: ConfigKey.SCREENSHOTS,
+    component: EuiButtonGroup,
+    label: i18n.translate('xpack.synthetics.monitorConfig.screenshotOptions.label', {
+      defaultMessage: 'Screenshot options',
+    }),
+    helpText: i18n.translate('xpack.synthetics.monitorConfig.screenshotOptions.helpText', {
+      defaultMessage: 'Set this option to manage the screenshots captured by the synthetics agent.',
+    }),
+    controlled: true,
+    props: ({
+      field,
+      setValue,
+    }: {
+      field?: ControllerRenderProps;
+      setValue: UseFormReturn['setValue'];
+    }) => ({
+      type: 'single',
+      idSelected: field?.value,
+      onChange: (option: ScreenshotOption) => setValue(ConfigKey.SCREENSHOTS, option),
+      options: Object.values(ScreenshotOption).map((option) => ({
+        id: option,
+        label: option.replace(/-/g, ' '),
+      })),
+      css: {
+        'text-transform': 'capitalize',
+      },
+    }),
+  },
+  [ConfigKey.TEXT_ASSERTION]: {
+    fieldKey: ConfigKey.TEXT_ASSERTION,
+    component: EuiFieldText,
+    label: i18n.translate('xpack.synthetics.monitorConfig.textAssertion.label', {
+      defaultMessage: 'Text assertion',
+    }),
+    required: true,
+    helpText: i18n.translate('xpack.synthetics.monitorConfig.textAssertion.helpText', {
+      defaultMessage: 'Consider the page loaded when the specified text is rendered.',
+    }),
+    validation: () => ({
+      required: true,
+    }),
+  },
+  [ConfigKey.THROTTLING_CONFIG]: {
+    fieldKey: ConfigKey.THROTTLING_CONFIG,
+    component: EuiSuperSelect,
+    label: i18n.translate('xpack.synthetics.monitorConfig.throttling.label', {
+      defaultMessage: 'Connection profile',
+    }),
+    required: true,
+    controlled: true,
+    helpText: i18n.translate('xpack.synthetics.monitorConfig.throttling.helpText', {
+      defaultMessage:
+        'Simulate network throttling (download, upload, latency). More options will be added in a future version.',
```

Comment:
Is that second sentence necessary? I haven't seen any references to future features elsewhere. 

```suggestion
        'Simulate network throttling (download, upload, latency).',
```